merino-adm/src/remote_settings.rs

```diff
@@ -60,53 +53,47 @@ impl RemoteSettingsSuggester {
-                .collection_name(&config.collection)
-                .storage(Box::new(FileStorage {
-                    folder: settings.remote_settings.storage_path.clone(),
-                    ..Default::default()
-                }));
-            if let Some(server) = &settings.remote_settings.server {
-                rs_client_builder = rs_client_builder.server_url(server);
-            }
-            rs_client_builder
-                .build()
-                .context("Creating RemoteSettings client")
-                .map_err(SetupError::InvalidConfiguration)?
-        };
+        // Get the base URL to download attachments from.
+        let remote_settings_server_info =
+            RemoteSettingsServerInfo::fetch(settings, &reqwest_client).await?;
+        let attachment_base_url = remote_settings_server_info.attachment_base_url()?;
+
+        // Get records from Remote Settings.
+        let records = {
+            let records_url = format!(
+                "{}/v1/buckets/{}/collections/{}/records",
+                settings.remote_settings.server, config.bucket, config.collection
+            );
 
-        // `.sync()` blocks while doing IO
-        rs_client
-            .sync(None)
-            .context("Syncing suggestions from remote settings")
-            .map_err(SetupError::Network)?;
+            let mut all_records = Vec::new();
+            let mut next_url = Some(records_url);
+            while let Some(url) = next_url {
+                let records_res = reqwest_client
+                    .get(&url)
+                    .send()
+                    .await
+                    .and_then(|res| res.error_for_status())
+                    .context(format!("Fetching records from remote settings: {}", url))
+                    .map_err(SetupError::Network)?;
 
-        // Get the base URL to download attachments from.
-        let attachment_base_url = &REMOTE_SETTINGS_SERVER_INFO
-            .get_or_try_init(|| RemoteSettingsServerInfo::fetch(&reqwest_client))
-            .await?
-            .attachment_base_url()?;
-
-        // Get records from Remote Settings, and convert them into a schema instead of using JSON `Value`s.
-        let records: Vec<SuggestRecord> = rs_client
-            // `.get()` blocks while doing IO
-            .get()
-            .context("Fetching records from remote settings")
-            .map_err(SetupError::Network)?
-            .into_iter()
-            .filter(|r| !r.deleted())
-            .map(|r| {
-                let value = Value::Object(r.as_object().clone());
-                <SuggestRecord as Deserialize>::deserialize(value)
-            })
-            .collect::<Result<_, <Value as serde::Deserializer>::Error>>()
-            .context("Parsing suggestions records")
-            .map_err(SetupError::Format)?;
+                next_url = match records_res.headers().get("Next").map(|v| v.to_str()) {
```

Comment:
Is this raw kinto being hit here?

I don't any references to a "Next" header but I see [a "Next-Page"](https://docs.kinto-storage.org/en/latest/api/1.x/pagination.html?highlight=Next-Page)

Reply:
Yes this is hitting Kinto directly. You're right. Dunno how I got just plan `Next`.